tests/test_pytesseract.py

```diff
@@ -1,25 +1,246 @@
+
+    if extension is 'pdf':
+        if IS_PYTHON_2:
+            assert isinstance(result, str)
+            result = str(result).strip()
+            assert result.startswith('%PDF')
+            assert result.endswith('EOF')
+        else:
+            assert isinstance(result, bytes)
+
+    if extension is 'hocr':
+        assert isinstance(result, bytes)  # type
+        result = result.decode('utf-8') if IS_PYTHON_2 else str(result, 'utf-8')
+        result = str(result).strip()
+        assert result.startswith('<?xml')
+        assert result.endswith('</html>')
+
+
+@pytest.mark.skipif(
+    TESSERACT_VERSION[:2] >= (3, 5),
+    reason='requires tesseract < 3.05'
+)
+def test_image_to_data__pandas_support(test_file):
+    with pytest.raises(TSVNotSupported):
+        image_to_data(test_file, output_type=Output.DATAFRAME)
+
+
+@pytest.mark.skipif(
+    TESSERACT_VERSION[:2] < (3, 5),
+    reason='requires tesseract >= 3.05'
+)
+@pytest.mark.skipif(
+    pandas is None,
+    reason='requires pandas'
+)
+def test_image_to_data__pandas_output(test_file):
+    """Test and compare the type and meta information of the result."""
+    result = image_to_data(test_file, output_type=Output.DATAFRAME)
+    assert isinstance(result, pandas.DataFrame)
+    expected_columns = ['level', 'page_num', 'block_num', 'par_num',
+                        'line_num', 'word_num', 'left', 'top', 'width',
+                        'height', 'conf', 'text']
+    assert bool(set(result.columns).intersection(expected_columns))
+
+
+@pytest.mark.skipif(
+    TESSERACT_VERSION[:2] < (3, 5),
+    reason='requires tesseract >= 3.05'
+)
+@pytest.mark.parametrize('output', [
+    Output.BYTES,
+    Output.DICT,
+    Output.STRING,
+], ids=[
+    'bytes',
+    'dict',
+    'string',
+])
+def test_image_to_data_common_output(test_file, output):
+    """Test and compare the type of the result."""
+    result = image_to_data(test_file, output_type=output)
+    expected_keys = ['level', 'page_num', 'block_num', 'par_num',
+                     'line_num', 'word_num', 'left', 'top', 'width',
+                     'height', 'conf', 'text']
+
+    if output is Output.BYTES:
+        assert isinstance(result, bytes)
+
+    elif output is Output.DICT:
+        assert isinstance(result, dict)
+        assert bool(set(result.keys()).intersection(expected_keys))
+
+    elif output is Output.STRING:
+        assert isinstance(result, unicode if IS_PYTHON_2 else str)
+        for key in expected_keys:
+            assert key in result
+
+
+@pytest.mark.parametrize('obj', [1, 1., None], ids=['int', 'float', 'none'])
+def test_wrong_prepare_type(obj):
+    with pytest.raises(TypeError):
+        prepare(obj)
+
+
+@pytest.mark.parametrize('path', [
+    r'wrong_tesseract',
+    r'',
+    os.path.sep + r'wrong_tesseract',
+    ], ids=[
+    'executable_name',
+    'empty_name',
+    'absolute_path',
+])
+def test_wrong_tesseract_cmd(test_file, path):  # This must be the last test!
```

Comment:
Does this need to be the last because it monkey patches `pytesseract.pytesseract.tesseract_cmd`?

If so, could the test restore the correct value?

Reply:
Good idea 👍 .

Reply:
It works!